### PR TITLE
Add connection between event and its state

### PR DESCRIPTION
**Pull request by iEvgeny**
Implemented the state displaying for the selected event on the event panel.

### CODE REVIEW

File: web/app/view/StateController.js [hunk 1]
```diff
@@ -175,6 +176,14 @@ Ext.define('Traccar.view.StateController', {
         }
     },
 
+    selectEvent: function (position) {
```
Comment (tananaev) [inline web/app/view/StateController.js]: Why do we need a duplicate method?

Comment (iEvgeny) [reply web/app/view/StateController.js]: As always, a little hurried...
Now it has been tested more thoroughly and I think it’s ready for a review.

File: web/app/view/map/MapMarkerController.js [hunk 2]
```diff
@@ -496,12 +493,13 @@ Ext.define('Traccar.view.map.MapMarkerController', {
                 this.reportMarkers[position.get('id')] = this.addReportMarker(position);
             }
             this.selectMarker(this.reportMarkers[position.get('id')], center);
+        } else if (this.selectedMarker) {
+            this.selectMarker(null, false);
```
[the remark this answers is not in the document]
Comment (iEvgeny) [reply web/app/view/map/MapMarkerController.js]: Could you please clarify what the problem is?

File: web/app/view/EventsController.js [hunk 3]
```diff
@@ -79,7 +83,14 @@ Ext.define('Traccar.view.EventsController', {
         Traccar.app.showEvents(false);
     },
 
-    deselectEvent: function () {
+
+    deselectEvent: function (object) {
+        if (object !== undefined) {
```
[the remark this answers is not in the document]
Comment (iEvgeny) [reply web/app/view/EventsController.js]: Fixed

File: web/app/view/ReportController.js [hunk 4]
```diff
@@ -251,6 +246,12 @@ Ext.define('Traccar.view.ReportController', {
         }
     },
 
+    deselectReport: function (object) {
+        if (object !== undefined) {
```
[the remark this answers is not in the document]
Comment (iEvgeny) [reply web/app/view/ReportController.js]: Fixed

File: web/app/view/map/MapMarkerController.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tananaev) [reply web/app/view/map/MapMarkerController.js]: I just don't understand the reason for recursion here. Can you explain.

[the remark this answers is not in the document]
Comment (iEvgeny) [reply web/app/view/map/MapMarkerController.js]: But there's no recursion...
`this.selectedMarker != this.selectMarker`

[the remark this answers is not in the document]
Comment (tananaev) [reply web/app/view/map/MapMarkerController.js]: Sorry, ignore my comment. I missed the code collapsing here.

**Comment by iEvgeny** [comment]
Made a small fix for the case of self-selection of the last unselected marker.
In my case, the "itemkeyup" event was sometimes called for the device grid, when the browser received input focus, which in turn led to an undesirable effect. Ext.grid.Panel has a "selectionchange" event which replaces "rowclick"/ "itemkeyup" events and behaves as expected.

File: web/app/view/edit/DevicesController.js [hunk 5]
```diff
@@ -111,9 +111,9 @@ Ext.define('Traccar.view.edit.DevicesController', {
     },
 
     onSelectionChange: function (el, record) {
-        if (record) {
-            this.updateButtons([record]);
-            this.fireEvent('selectdevice', record, true);
+        if (record && record.length) {
```
Comment (tananaev) [inline web/app/view/edit/DevicesController.js]: The variable name should change, if it's an array.

**Comment by iEvgeny** [comment]
Of course. Fixed.

ср, 27 февр. 2019 г. в 20:51, Anton Tananaev <notifications@github.com>:
>
> @tananaev commented on this pull request.
>
> ________________________________
>
> In web/app/view/edit/DevicesController.js:
>
> > @@ -111,9 +111,9 @@ Ext.define('Traccar.view.edit.DevicesController', {
>      },
>
>      onSelectionChange: function (el, record) {
> -        if (record) {
> -            this.updateButtons([record]);
> -            this.fireEvent('selectdevice', record, true);
> +        if (record && record.length) {
>
> The variable name should change, if it's an array.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.



-- 
С уважением, Евгений!
Тел. +7 (951) 916-58-73

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**